tcp.go

```diff
@@ -33,6 +33,20 @@ func tryLinger(conn net.Conn, sec int) {
 	}
 }
 
+func tryKeepAlive(conn net.Conn, keepAlive bool) {
+	type canKeepAlive interface {
+		SetKeepAlive(bool) error
+	}
+
+	var _ canKeepAlive = &net.TCPConn{}
```

Comment:
why the initializer here?

Reply:
To check that the `net.TCPConn` actually satisfies the `canKeepAlive` interface. This should be a trivial thing to remove for the compiler, so there's no performance impact.

Reply:
Maybe just move this interface/var out of the function? That'll make it clear what's happening.